### PR TITLE
Feat: add seedoss

### LINKED ISSUE #3097: Seed-OSS-36B Support
### ⚠️ Please check that this feature request hasn't been suggested before.

- [x] I searched previous [Ideas in Discussions](https://github.com/axolotl-ai-cloud/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [x] I searched previous [Issues](https://github.com/axolotl-ai-cloud/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Is it possible to add support for https://huggingface.co/ByteDance-Seed/Seed-OSS-36B-Instruct

This is a new dense model with both instruct and multiple base versions available that looks quite promising.

### ✔️ Solution

Axolotl support for finetuning Seed-OSS models.

### ❓ Alternatives

_No response_

### 📝 Additional Context

_No response_

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this feature has not been requested yet.
- [x] I have provided enough information for the maintainers to understand and evaluate this request.

### PR DESCRIPTION
**Pull request by NanoCode012**
# Description

Final step on example config
```
{'loss': 1.1177, 'grad_norm': 0.06398500502109528, 'learning_rate': 2.5072087818176382e-06, 'memory/max_mem_active(gib)': 27.13, 'memory/max_mem_allocated(gib)': 27.13, 'memory/device_mem_reserved(gib)': 27.72, 'epoch': 1.0}
```

TODO:
- [x] Run single gpu test for usage for doc
- [ ] make multi-gpu example

Closes #3097 

## Summary by CodeRabbit

- New Features
  - Added a complete Seed-OSS 36B QLoRA finetuning example: config, step-by-step README, VRAM guidance, dataset/inference tips, and optimization notes.

- Documentation
  - Expanded Cut Cross Entropy docs to list three new supported models (deepseek_v3, glm4_moe, seed_oss) and updated installation instructions.

- Chores
  - Aligned the referenced ml-cross-entropy commit across examples, docs, and helper scripts; added seed_oss to supported multipack model types.

### CODE REVIEW

File: examples/seed-oss/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+This guide shows how to fine-tune it with Axolotl with multi-turn conversations and proper masking.
+
+## Getting started
+
+1. Install Axolotl following the [installation guide](https://docs.axolotl.ai/docs/installation.html). You need to install from main as Seed-OSS is only on nightly or use our latest [Docker images](https://docs.axolotl.ai/docs/docker.html).
+
+    Here is an example of how to install from main for pip:
+
+```bash
+# Ensure you have Pytorch installed (Pytorch 2.6.0 min)
```
Comment (salmanmohammadi) [inline examples/seed-oss/README.md:14]: Unrelated but we should try keep a single source of truth for installation from main and just point to it

File: examples/seed-oss/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+# Install Cut Cross Entropy
+python scripts/cutcrossentropy_install.py | sh
+```
+
+2. Install Seed-OSS transformers commit:
+
+```bash
+pip3 uninstall -y transformers
+
+pip3 install git+https://github.com/huggingface/transformers.git@56d68c6706ee052b445e1e476056ed92ac5eb383
```
Comment (salmanmohammadi) [inline examples/seed-oss/README.md]: Do we need a `--no-deps`?

**Review by salmanmohammadi — APPROVED** [review]
bueno

File: examples/seed-oss/README.md (hunk 1 above)
Comment (NanoCode012) [reply examples/seed-oss/README.md:14]: Thanks. I always link the main docs but thought having a pip section would make it one step easier. Should we just get rid of this pip section?

File: examples/seed-oss/README.md (hunk 2 above)
Comment (NanoCode012) [reply examples/seed-oss/README.md]: It should be fine, I tested using this same cli on runpod. We can remove this section once into a release upstream.
